### PR TITLE
Remove a bunch of just-about-dead Ember message send code.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
sendToClient in ias-zone-server is the only place all this machinery
could be reached from in practice.  And since at that one callsite the
machinery is not going to do anything useful, there's no point in
pretending like it works and keeping it.

#### Problem
Lots of pretty-much-dead code.

#### Change overview
Remove it.  Save some codesize and rodata and .bss.

#### Testing
Should be no behavior changes except in the IAS bits, which don't work right anyway and need fixing.

### CODE REVIEW

File: src/app/util/client-api.h [hunk 1]
```diff
@@ -94,53 +94,6 @@
 uint16_t emberAfFillExternalBuffer(uint8_t frameControl, chip::ClusterId clusterId, chip::CommandId commandId, const char * format,
```
Comment (mrjerryjohns) [inline src/app/util/client-api.h:94]: Can't we not delete this?

File: src/app/util/client-api.h [hunk 2]
```diff
@@ -152,16 +105,6 @@ uint16_t emberAfFillExternalManufacturerSpecificBuffer(uint8_t frameControl, chi
  */
 void emberAfSetExternalBuffer(uint8_t * buffer, uint16_t bufferLen, uint16_t * responseLengthPtr, EmberApsFrame * apsFramePtr);
```
Comment (mrjerryjohns) [inline src/app/util/client-api.h:106]: Why not pretty much everything in this file?

File: src/app/util/client-api.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/util/client-api.h:94]: We probably can, but it had a number of callsites in cluster code we compile (messaging-server and ias-zone-server) and I did not want to scope-creep this PR any more than it already had....

But yes, those callsites are obviously buggy and we should do something with them and delete this function.

File: src/app/util/client-api.h (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/util/client-api.h:106]: Honestly?  I got tired of adding more stuff to this PR and needed to do other things.  Will do followups for more removals.

**Review by tcarmelveilleux — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Pull request merged**